src/inmanta/module.py

```diff
@@ -1397,19 +1459,80 @@ def require_v2(module_name: str) -> None:
                 # already v2
                 return
             v2_modules.add(module_name)
-            if module_name in done and len(done[module_name]) > 0:
+            if module_name in set_up:
+                set_up.remove(module_name)
+            if module_name in done:
                 # some submodules already loaded as v1 => reload
                 imports.extend(done[module_name].values())
                 del done[module_name]
-                del ast_by_top_level_mod[module_name]
+                if module_name in ast_by_top_level_mod:
+                    del ast_by_top_level_mod[module_name]
 
+        def load_module_requirements(module_like: ModuleLike) -> None:
```

Comment:
```suggestion
        def load_v2_module_requirements(module_like: ModuleLike) -> None:
```

Reply:
I've already renamed this to `load_module_v2_requirements` following a comment by @wouterdb. I assume that's fine as well?